dot/network/notifications.go

```diff
@@ -158,6 +158,8 @@ func (s *Service) createNotificationsMessageHandler(
 
 			hs, ok := msg.(Handshake)
 			if !ok {
+				// this error will never happen.
+				// Handshake interface and NotificationMessage interfaces are same.
```

Comment:
since both interfaces are the same, what if:
```
var hs Handshake = msg
```
is that possible? if so, we don't need to cast and don't need to check the `ok` value

Reply:
I'd just use `handshake, ok := msg.(Handshake)` instead of `msg.IsHandshake()` + if condition, right?

Reply:
it is not necessary to use the cast + `ok` check if both `Handshake` and `NotificationMessage` are the same

Reply:
> is that possible? if so, we don't need to cast and don't need to check the ok value

Maybe, but I don't see the point of that change. The reason I put that comment is to point out a problem. The solution to that is larger than one fix or so.

Reply:
> I'd just use handshake, ok := msg.(Handshake) instead of msg.IsHandshake() + if condition, right?

` handshake, ok := msg.(Handshake) ` is not enough to check, because it will always be ok, because they follow the same interface.

Reply:
I understand you want to point out a problem that need to be addressed in a future larger refactor, but my point is: the cast + ok check is a useless code and we should avoid it, you can point out the problem using a `//TODO` comment that points to an issue and then pass the `msg` variable to the `info.handshakeValidator(peer, msg)` without casting to `Handshake` interface:

```go
err := info.handshakeValidator(peer, msg)
```
instead of

```go
hs, ok := msg.(Handshake)
if !ok {
  // this error will never happen.
  // Handshake interface and NotificationMessage interfaces are same.
  return errMessageIsNotHandshake
}

...
err := info.handshakeValidator(peer, hs)
```

Reply:
What you are saying makes sense. I can not cast it and I can directly pass a msg. But, when we don that we make the assumption that Handshake and NotificationMessage have the same interface. Which they have. But what happen when in future they are not the same interface?

If I were to directly pass the msg, I might as well remove the Handshake interface. But, I don't think we should remove handshake interface. Theoretically, we can remove Handshake interface, but Handshake interface makes the code a lot more readable. It tells us, when handshakes are getting used.